src/services/muutoshakemus/utils/changes-to-save.js

```diff
@@ -500,17 +501,32 @@ export function getChangesToSave(
         perustelutForBackend,
         perusteluteksti ? { perusteluteksti } : null
       );
-      return {
+
+      let muutokset = [{
         arvo: changeObj.properties.applyForValue,
         kategoria: R.head(anchorParts),
         koodiarvo,
         koodisto: koodisto.koodistoUri,
         kohde,
-        maaraystyyppi: R.find(R.propEq("tunniste", "OIKEUS"), maaraystyypit),
+        maaraystyyppi: isVahimmaismaara
+            ? R.find(R.propEq("tunniste", "OIKEUS"), maaraystyypit)
+            : R.find(R.propEq("tunniste", "RAJOITE"), maaraystyypit),
         meta,
-        tila: "MUUTOS",
-        type: "change"
-      };
+        tila: "LISAYS",
+        type: "addition"
+      }];
+
+      // Add removal change if old maarays exists
+      if (changeObj.properties.metadata.koodiarvo.maaraysUuid) {
+        muutokset = R.append({
+          ...muutokset[0],
+          maaraysUuid: changeObj.properties.metadata.koodiarvo.maaraysUuid,
```

Comment:
metadata.koodiarvo should be renamed because it clearly includes other properties that koodiarvo related ones

Reply:
Good point 👍